pkg/typesystem/weighted_graph_test.go

```diff
@@ -270,81 +232,61 @@ func TestGetEdgesForIntersection(t *testing.T) {
 		require.True(t, ok)
 		comparator, err := GetEdgesForIntersection(actualIntersectionEdges, "user")
 		require.NoError(t, err)
-		require.Equal(t, IntersectionEdges{}, comparator)
+		require.Len(t, comparator.LowestEdges, 1)
+		require.Equal(t, graph.TTUEdge, comparator.LowestEdges[0].GetEdgeType())
+		require.Equal(t, "subteam#member", comparator.LowestEdges[0].GetTo().GetUniqueLabel())
+		require.Len(t, comparator.SiblingEdges, 1)
+		require.Equal(t, graph.RewriteEdge, comparator.SiblingEdges[0][0].GetEdgeType())
+		require.Equal(t, graph.IntersectionOperator, comparator.SiblingEdges[0][0].GetTo().GetLabel())
 	})
-
-	t.Run("nested_operator", func(t *testing.T) {
+	t.Run("error_non_intersection_node", func(t *testing.T) {
 		model := `
-		model
-			schema 1.1
-		type user
-		type subteam
-			relations
-				define member: [user]
-		type adhoc
-			relations
-				define member: [user]
-		type team
-			relations
-				define member: [subteam#member]
-		type group
-			relations
-				define team: [team]
-				define subteam: [subteam]
-				define adhoc_member: [adhoc#member]
-				define member: (member from team and adhoc_member) and member from subteam
-		`
+			model
+				schema 1.1
+			type user
+			type user2
+			type group
+				relations
+					define allowed: [user]
+					define member: [user] and allowed
+					define other: [user2]
+			`
 		typeSystem, err := New(testutils.MustTransformDSLToProtoWithID(model))
 		require.NoError(t, err)
+		currentNode, ok := typeSystem.authzWeightedGraph.GetNodeByID("group#other")
+		require.True(t, ok)
 
-		edges, _, err := typeSystem.GetEdgesForListObjects("group#member", "user")
-		require.NoError(t, err)
-
-		require.Len(t, edges, 1)
-		rootIntersectionNode := edges[0].GetTo()
-		require.Equal(t, graph.IntersectionOperator, rootIntersectionNode.GetLabel())
-		require.Equal(t, graph.OperatorNode, rootIntersectionNode.GetNodeType())
-		actualIntersectionEdges, ok := typeSystem.authzWeightedGraph.GetEdgesFromNode(rootIntersectionNode)
+		actualIntersectionEdges, ok := typeSystem.authzWeightedGraph.GetEdgesFromNode(currentNode)
 		require.True(t, ok)
-		comparator, err := GetEdgesForIntersection(actualIntersectionEdges, "user")
-		require.NoError(t, err)
-		require.False(t, comparator.DirectEdgesAreLeastWeight)
-		require.NotNil(t, comparator.LowestEdge)
-		require.Equal(t, graph.TTUEdge, comparator.LowestEdge.GetEdgeType())
-		require.Equal(t, "subteam#member", comparator.LowestEdge.GetTo().GetUniqueLabel())
-		require.Empty(t, comparator.DirectEdges)
-		require.Len(t, comparator.Siblings, 1)
-		require.Equal(t, graph.OperatorNode, comparator.Siblings[0].GetTo().GetNodeType())
-		require.Equal(t, graph.RewriteEdge, comparator.Siblings[0].GetEdgeType())
-		require.Equal(t, graph.IntersectionOperator, comparator.Siblings[0].GetTo().GetLabel())
+		_, err = GetEdgesForIntersection(actualIntersectionEdges, "user")
+		require.Error(t, err)
 	})
-	t.Run("non_directed_assignment_no_path", func(t *testing.T) {
+
```

Comment:
Can we also add a case where there are multiple TTUs in an intersection and this is the lowest weight? i.e.,

```
			model
				schema 1.1
			type user
			type subteam1
				relations
					define member: [user]
			type subteam2
				relations
					define member: [user]
			type adhoc
				relations
					define member: [subteam1#member]
			type group
				relations
					define parent: [subteam1, subteam2]
					define member: [adhoc#member] and member from parent
```